vendor/github.com/optiopay/kafka/proto/messages.go

```diff
@@ -113,12 +113,22 @@ func ReadReq(r io.Reader) (requestKind int16, b []byte, err error) {
 	if err != nil {
 		return 0, nil, err
 	}
+
 	binary.BigEndian.PutUint32(b, uint32(msgSize))
-	binary.BigEndian.PutUint16(b[4:], uint16(requestKind))
-	if _, err := io.ReadFull(r, b[6:]); err != nil {
-		return 0, nil, err
+
+	// only write back requestKind if it was included in messageSize
+	if len(b) >= 6 {
+		binary.BigEndian.PutUint16(b[4:], uint16(requestKind))
 	}
-	return requestKind, b, err
+
+	// read rest of request into allocated buffer if we allocated for it
+	if len(b) > 6 {
```

Comment:
@tgraf How does this assure that we read the correlation ID field too? afaict, this takes care until the apiKey of the header only?

Reply:
This ensures that the parser doesn't crash, if the message size does not cover the correlation ID then it is not safe to read it.